File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        int n = elements.length;
        Set<Integer> set = new HashSet<>();
        
        for (int size = 1; size < n + 1; size++) {
            int sum = 0;
            for (int i = 0; i < size; i++) {
                sum += elements[i];
            }
            set.add(sum);
            for (int i = 0; i < n; i++) {
                sum -= elements[i];
                sum += elements[(i + size) % n];
                set.add(sum);
            }
        }
        return set.size();
    }
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    n = len(elements)
    answer = set()
    for size in range(1, n + 1):
        cur_sum = sum(elements[:size])
        answer.add(cur_sum)
        for i in range(n):
            cur_sum -= elements[i]
            cur_sum += elements[(i + size) % n]
            answer.add(cur_sum)
    return len(answer)